thrall/app/controllers/HealthCheck.scala

```diff
@@ -1,17 +1,35 @@
 package controllers
 
-import play.api.mvc.{Action, Controller}
+import play.api.mvc.{Result, Action, Controller}
 import play.api.libs.concurrent.Execution.Implicits._
-import lib.{MessageConsumer, ElasticSearch}
+import lib.{Config, MessageConsumer, ElasticSearch}
 import com.gu.mediaservice.syntax._
 
 object HealthCheck extends Controller {
 
   def healthCheck = Action.async {
+    elasticHealth map {
+      case r: Result => sqsHealth
+      case _ => ServiceUnavailable("ES is not healthy")
+    }
+  }
+
+  def elasticHealth = {
     ElasticSearch.client.prepareSearch().setSize(0)
       .executeAndLog("Health check")
       .filter(_ => ! MessageConsumer.actorSystem.isTerminated)
-      .map(_ => Ok("OK"))
+      .map(_ => Ok("ES is healthy"))
   }
 
+  def sqsHealth = {
+    val beatsSinceLastCheck = MessageConsumer.heartbeats.getAndSet(0)
+    val lastHeartBeat = MessageConsumer.lastHeartBeat.get
+
+    if (beatsSinceLastCheck == 0)
+      ServiceUnavailable("Heart has not beat since last health check")
```

Comment:
What is this check for? Does that mean GETing the healthcheck endpoint results in a state mutation on the server?